### PR TITLE
Adding additional cloud commands to CLI

### PR DESCRIPTION
**Pull request by dsiddharth**
https://github.com/hathora/hathora/pull/274/files

### CODE REVIEW

File: cli.ts [hunk 1]
```diff
@@ -1,16 +1,39 @@
...
 
 updateNotifier({ pkg: require("./package.json") }).notify({ defer: false, isGlobal: true });
 
+const cloudMiddleware: MiddlewareFunction = (argv) => {
+  if (argv._[0] !== "cloud") {
+    return;
+  }
+
+  if (!(argv._[1] === "login" || "token" in argv)) {
+    const tokenFile = join(os.homedir(), ".config", "hathora", "token");
```
Comment (hpx7) [inline cli.ts:20]: for a future PR: maybe `tokenDir` should be part of `getDirs`

File: commands/cloud/deploy.ts [hunk 2]
```diff
@@ -29,20 +31,16 @@ const cmd: CommandModule = {
...
+    const headers = { Authorization: `Bearer ${argv.token}` };
+    try {
+      const response = await axios.postForm(`${argv.cloudApiBase}/deploy`, form, { headers, responseType: "stream" });
+      response.data.pipe(process.stdout);
+    } catch (err) {
+      if (axios.isAxiosError(err)) {
+        (err.response?.data as Stream).on("data", (data) => console.log(data.toString()));
       }
-    });
+      console.error(err);
```
Comment (hpx7) [inline commands/cloud/deploy.ts]: this should be in an else

File: utils.ts [hunk 3]
```diff
@@ -9,6 +9,28 @@ import shelljs from "shelljs";
...
+      responseType: "stream",
+    });
+
+    response.data.on('data', (d: any) => process.stdout.write(d));
+    response.data.on('end', () => process.stdout.write("\n"));
+  } catch (err) {
+    if (axios.isAxiosError(err)) {
+      (err.response?.data as Stream).on("data", (data) => console.error(data.toString()));
+    }
+    console.error(err);
```
Comment (hpx7) [inline utils.ts]: same for this

File: utils.ts [hunk 4]
```diff
@@ -9,6 +9,28 @@ import shelljs from "shelljs";
 import { createServer } from "vite";
 import chalk from "chalk";
 import { generate } from "./generate";
+import axios, { Method } from "axios";
+import { Stream } from "stream";
```
Comment (hpx7) [inline utils.ts:13]: since this is a node stdlib import, can you put it up next to it others (like right below `crypto`)?

Comment (dsiddharth) [reply utils.ts:13]: i'm going to do the import sorting through linting soon

**Review by hpx7 — APPROVED** [review]

**Pull request merged**